AlfrescoApp/Model/Managers/RealmSyncManager.m

```diff
@@ -1300,7 +1300,20 @@ - (void)rebuildDataBaseWithCompletionBlock:(void (^)(BOOL completed))completionB
         }
         else
         {
-            [self handleNodeSyncActionAndStatus:node.alfrescoNode parentNode:nil];
+            //document service get node for id
+            [self.documentFolderService retrieveNodeWithIdentifier:rnode.alfrescoNode.identifier
+                                                   completionBlock:^(AlfrescoNode *node, NSError *error) {
+                if (error)
+                {
+                    [self removeTopLevelNodeFlagFomNodeWithIdentifier:[[RealmSyncCore sharedSyncCore]
+                                                                       syncIdentifierForNode:rnode.alfrescoNode]];
```

Comment:
Calling .alfrescoNode inside a block could create some problems. It would be better to capture the alfrescoNode into a variable outside of the block and then use it inside the block.